src/js/controller/captions.js

```diff
@@ -130,10 +132,10 @@ const Captions = function(_model) {
                 // TODO: auto select track by comparing track.language to system lang
             }
         }
-        // set the index without the side effect of storing the Off label in _selectCaptions
-        if (_model.get('enableDefaultCaptions') && _model.get('autostart') && !captionsMenuIndex) {
+        if (_captionsByDefault && !captionsMenuIndex) {
```

Comment:
`captionsByDefault`

Reply:
Wouldn't it be great if this was the suggested index?

```js 
// when enableDefaultCaptions is set
captions.selectDefaultIndex(1);

// remember the default index
function _selectDefaultIndex(defaultIndex) {
  _defaultIndex = defaultIndex;

// set the index to the default
if (defaultIndex && !captionsMenuIndex) {
  captionsMenuIndex = defaultIndex;
```

Reply:
`
_selectDefaultIndex(defaultIndex) {
   _defaultIndex = defaultIndex;
`
I didn't to put the saving logic here, because `_selectDefaultIndex` is called with empty param on `_setCaptionsList` function (called on itemReady), and autostart checking logic only happens once when player changes from idle state.

Instead, it is placed in the `this.selectDefaultIndex`, which should be the only case where the default index changes.

Reply:
```js
// when enableDefaultCaptions is set
captions.selectDefaultIndex(1);

// everywhere else that  _selectDefaultIndex in this module:
_selectDefaultIndex(_defaultIndex);

// remember the default index
function _selectDefaultIndex(defaultIndex) {
  _defaultIndex = defaultIndex;

// set the index to the default
if (defaultIndex && !captionsMenuIndex) {
  captionsMenuIndex = defaultIndex;
```